ibis/backends/oracle/__init__.py

```diff
@@ -192,8 +192,16 @@ def _from_url(self, url: ParseResult, **kwargs):
         return self
 
     @property
-    def current_database(self) -> str:
+    def current_catalog(self) -> str:
         with self._safe_raw_sql(sg.select(STAR).from_("global_name")) as cur:
+            [(catalog,)] = cur.fetchall()
+        return catalog
+
+    @property
+    def current_database(self) -> str:
+        # databases correspond to users, other than that there's
+        # no notion of a database inside a catalog for oracle
```

Comment:
So given that there's no notion of a database inside a catalog, should we call this `database`?

If a thing is a collection of tables, e.g. `IBIS_TESTING`, then that should be called `database`.

Reply:
well, you can definitely specify tables like this:

```sql
CREATE TABLE $MY_USER.$MY_TABLE
```

so it seems like it fits the bill in practice, if not nominally.

Reply:
`IBIS_TESTING` has no single noun attached to it AFAICT.

The documentation for `global_name` uses the word "database" to describe it, while on `self.con` the return value of `self.con.service_name` is `"IBIS_TESTING"`.

Reply:
I had happily forgotten about `service_name`.

Hmm, so we can _also_ specify things this way:
```python
[ins] In [4]: con.table("ALL_DOMAINS", database="SYS")
Out[4]: 
DatabaseTable: SYS.ALL_DOMAINS
  OWNER            !string
  NAME             !string
  COLS             !int64
  DISCRIMINANTCOLS int64
  BUILTIN          boolean
  DATA_DISPLAY     string
  DISPLAY_LENGTH   int64
  DATA_ORDER       string
  ORDER_LENGTH     int64
  SELECTOR         string
  SELECTOR_LENGTH  int64
  TYPE             string
```

e.g. `SYS.ALL_DOMAINS` is a valid "database" in the Ibis terminology.

Oh wait, and I see that it shows up in `list_databases`, because we haven't changed that.

Ok, if the major change here is that we store the "service_name" (or whatever) in `current_catalog`, and there is no `list_catalogs` implemented (because it wouldn't make sense), and other collections of tables show up in `list_databases`, then I'm good with this.

Reply:
`SYS` [is actually a *user*](https://docs.oracle.com/en/database/oracle/oracle-database/19/admin/getting-started-with-database-administration.html#GUID-79AB6187-1522-4EB7-8FAD-E4322262AC65), so I think it fits what I've done here quite well.

Reply:
1. `list_catalogs` doesn't make sense AFAICT
2. `list_databases` *does* makes sense, and a "database" corresponds to a user (I believe this user-as-database pattern shows up in other systems as well).